bpf/trace_common.h

```diff
@@ -290,43 +290,43 @@ static __always_inline void get_or_create_trace_info(http_connection_metadata_t
     //make_tp_string(tp_buf, &tp_p->tp);
     //bpf_dbg_printk("tp: %s", tp_buf);
 
-#ifdef BPF_TRACEPARENT
-    // The below buffer scan can be expensive on high volume of requests. We make it optional
-    // for customers to enable it. Off by default.
-    if (!capture_header_buffer) {
-        bpf_map_update_elem(&trace_map, conn, tp_p, BPF_ANY);
-        server_or_client_trace(meta, conn, tp_p);
-        return;
-    }
+    if (BPF_TRACEPARENT) {
```

Comment:
I'm not sure this will work, I think on older kernels we'll fail to load the bpf program because the bpf_loop helper is still present in the code. Maybe the new compiler is somehow smarter, but I wonder if the same will work with older clang?

I think the best approach is to guard this statement with bpf_core_enum_value_exists, like we do for snprintf in dbg.h:

```
            if (bpf_core_enum_value_exists(enum bpf_func_id___x, BPF_FUNC_snprintf___x)) {         \
```

We don't need an additional constant, if the kernel has bpf_loop we'll look into the headers.

Reply:
We already have a constant for this if the constant is sufficient, e.g `capture_header_buffer`. But overall, this will simply rely on the userspace detecting the kernel, best is to check if the feature exists or not.

Reply:
Right, I should have tested on older kernels, the issue would have become apparent. Will rectify!